### DISCUSSION TITLE
Qdrant Architecture

### DISCUSSION THREAD

**Discussion by hritikakolkar** [qdrant/qdrant#1500] [General]
Is there any presentation that shows and explains qdrant neural search architecture? Actually, I am trying to understand the code but it's in rust. Please share any file so that I can understand the architecture.

**Comment by azayarni** [comment]
You do not have to understand the code. Qdrant is using HNSW under the hood, a special implementation with filtering support. https://qdrant.tech/articles/filtrable-hnsw/

PS: I highly encourage you to join our Discord server to get responses in minutes instead of weeks ;) https://qdrant.to/discord
